plugins/exporter.koplugin/target/memos.lua

```diff
@@ -0,0 +1,128 @@
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local http = require("socket.http")
+local json = require("json")
+local logger = require("logger")
+local ltn12 = require("ltn12")
+local socket = require("socket")
+local socketutil = require("socketutil")
+local _ = require("gettext")
+
+-- readwise exporter
+local MemosExporter = require("base"):new {
+    name = "memos",
+    is_remote = true,
+}
+
+local function makeRequest(method, request_body, api)
+    local sink = {}
+    local request_body_json = json.encode(request_body)
+    local source = ltn12.source.string(request_body_json)
+    socketutil:set_timeout(socketutil.LARGE_BLOCK_TIMEOUT, socketutil.LARGE_TOTAL_TIMEOUT)
+    local request = {
+        url     = api,
+        method  = method,
+        sink    = ltn12.sink.table(sink),
+        source  = source,
+        headers = {
+            ["Content-Type"] = "application/json"
+        },
+    }
+    local code, headers, status = socket.skip(1, http.request(request))
+    socketutil:reset_timeout()
+
+    if code ~= 200 then
+        logger.warn("Memos: HTTP response code <> 200. Response status:", status or code or "network unreachable")
+        logger.dbg("Response headers:", headers)
+        return nil, status
+    end
+
+    local response = json.decode(sink[1])
+    return response
+end
+
+function MemosExporter:isReadyToExport()
+    if self.settings.api then return true end
+    return false
+end
+
+function MemosExporter:getMenuTable()
+    return {
+        text = _("Memos"),
+        checked_func = function() return self:isEnabled() end,
+        sub_item_table = {
+            {
+                text = _("Set Memos API URL"),
+                keep_menu_open = true,
+                callback = function()
+                    local auth_dialog
+                    auth_dialog = InputDialog:new {
+                        title = _("Set API URL for Memos"),
+                        input = self.settings.api,
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
```

Comment:
Can you add `id = "close"` to enable exiting the dialog on non-touch devices.

Reply:
I am not familiar with lua and koreader, this is what I changed according to the code of other export plugins, now please see if there is any problem with my modification, thank you.